src/db/car/forest.rs

```diff
@@ -116,15 +116,18 @@ impl<ReaderT: super::RandomAccessFileReader> ForestCar<ReaderT> {
         let mut footer_buffer = [0; ForestCarFooter::SIZE];
         cursor.read_exact(&mut footer_buffer)?;
 
-        let footer = ForestCarFooter::try_from_le_bytes(footer_buffer).ok_or(invalid_data(
-            format!("not recognizable as a `{}` file", FOREST_CAR_FILE_EXTENSION),
-        ))?;
+        let footer = ForestCarFooter::try_from_le_bytes(footer_buffer).ok_or_else(|| {
+            invalid_data(format!(
+                "not recognizable as a `{}` file",
+                FOREST_CAR_FILE_EXTENSION
+            ))
+        })?;
 
         let cursor = Cursor::new_pos(&reader, 0);
         let mut header_zstd_frame = decode_zstd_single_frame(cursor)?;
         let block_frame = UviBytes::<Bytes>::default()
             .decode(&mut header_zstd_frame)?
-            .ok_or(invalid_data("malformed uvibytes"))?;
```

Comment:
```rust
#[stable(feature = "rust1", since = "1.0.0")]
    pub fn new<E>(kind: ErrorKind, error: E) -> Error
    where
        E: Into<Box<dyn error::Error + Send + Sync>>,
    {
        Self::_new(kind, error.into())
    }

fn _new(kind: ErrorKind, error: Box<dyn error::Error + Send + Sync>) -> Error {
        Error { repr: Repr::new_custom(Box::new(Custom { kind, error })) }
    }
```